### PR TITLE
Adding axial expansion support for CRAs

### PR DESCRIPTION
**Pull request by albeanth**
## Description
This PR adds support for axial expansion of pin-type control rod assemblies (CRAs). The primary assumption with this approach is that all components axially expand upwards. This is a valid assumption as long as the z-elevation of the control bundle tip is relative to itself; i.e., simply placing the bundle at a prescribed z-elevation. This is in contrast to the case where the z-elevation of the control bundle is relative to the control rod driveline. In this case, the downward expansion of components within the control bundle becomes important and this approach is invalidated. 

This PR will remain in draft state until 1) unit tests are added to support CRA expansion, and 2) the effects on downstream applications have been quantified. 

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/reactor/converters/axialExpansionChanger.py [hunk 1]
```diff
@@ -74,26 +74,18 @@ def expandColdDimsToHot(assems, isDetailedAxialExpansion, referenceAssembly=None
...
-            axialExpChanger.setAssembly(a)
-            # this doesn't get applied to control assems, so CR will be interpreted
-            # as hot. This should be conservative because the control rods will
-            # be modeled as slightly shorter with the correct hot density. Density
-            # is more important than height, so we are forcing density to be correct
-            # since we can't do axial expansion (yet)
-            axialExpChanger.applyColdHeightMassIncrease()
-            axialExpChanger.expansionData.computeThermalExpansionFactors()
-            axialExpChanger.axiallyExpandAssembly()
+        axialExpChanger.setAssembly(a)
```
[the remark this answers is not in the document]
Comment (jakehader) [reply armi/reactor/converters/axialExpansionChanger.py:77]: That is unless it's standard and "just works"

[the remark this answers is not in the document]
Comment (albeanth) [reply armi/reactor/converters/axialExpansionChanger.py:77]: It seems to be "just working". Though I'm still testing. I'll ping you directly.

**Review by keckler — CHANGES_REQUESTED** [review]
Easy enough. You need to add something to the changelog. I imagine this might change some results and confuse people.

**Review by keckler — APPROVED** [review]

**Comment by albeanth** [comment]
Coverage drop appears to be coverage spray....

**Comment by john-science** [comment]
@albeanth This is waiting on a downstream project.  So, short-term holding pattern.

**Comment by albeanth** [comment]
This PR is ready, but I am converting to draft while we figure out downstream merge dependencies.

**Pull request merged**